### PR TITLE
Support Windows Arm64 Build

### PR DESCRIPTION
**Pull request by shibayan**
Relate #1270 

A build for Windows on Arm will be added to Volta.

The changes I made include creating an msi and zip for Arm64 in my repository, and I have confirmed that it works on Windows on Arm machines.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -117,30 +148,40 @@ jobs:
           TAG: ${{ github.ref }}
         run: echo "version=${TAG:11}" >> $GITHUB_OUTPUT
       - name: Fetch Linux artifact
-        uses: actions/download-artifact@v2
+        uses: actions/download-artifact@v4
```
Comment (chriskrycho) [inline .github/workflows/release.yml:151]: Let's do this separately! I am happy to take care of that, though it will be this evening or later this week; and you can merge those changes in or rebase on top of them once we have!

Comment (shibayan) [reply .github/workflows/release.yml]: Thanks, I understand. In this PR, I revert the version change of the download-artifact and create a separate PR to address it.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -100,13 +100,44 @@ jobs:
...
+        with:
+          name: windows-zip-arm
+          path: target/aarch64-pc-windows-msvc/release/volta-windows-arm.zip
+
   release:
     name: Publish release
     runs-on: ubuntu-latest
     needs:
       - linux
+      - linux-arm
```
Comment (chriskrycho) [inline .github/workflows/release.yml:137]: I don’t think we should enable `linux-arm` in this PR. If we choose to enable it before shipping 2.0, we should do so as part of its own PR!

```suggestion
```

File: wix/main.wxs [hunk 3]
```diff
@@ -21,11 +21,9 @@
   destination for 32-bit versus 64-bit installers. Removal of these lines will
   cause installation errors.
 -->
-<?if $(var.Platform) = x64 ?>
-    <?define Win64 = "yes" ?>
+<?if $(sys.BUILDARCH) = x64 or $(sys.BUILDARCH) = arm64 ?>
     <?define PlatformProgramFilesFolder = "ProgramFiles64Folder" ?>
 <?else ?>
-  <?define Win64 = "no" ?>
```
Comment (chriskrycho) [inline wix/main.wxs:28]: Why the `Win64` changes here?

File: wix/main.wxs [hunk 4]
```diff
@@ -192,9 +191,11 @@
             <ComponentGroupRef Id='Binaries'/>
         </Feature>
 
+        <?if $(sys.BUILDARCH) = x64 ?>
         <Feature Id='VCRedistributable' Title='Visual C++ Runtime' AllowAdvertise='no' Display='hidden' Level='1'>
             <MergeRef Id='VCRedist'/>
         </Feature>
+        <?endif ?>
```
Comment (chriskrycho) [inline wix/main.wxs:198]: Likewise, why the addition of the `BUILDARCH` check here?

File: wix/main.wxs [hunk 5]
```diff
@@ -44,12 +42,11 @@
             Keywords='Installer'
             Description='The JavaScript Launcher'
             Manufacturer='The Volta Maintainers'
-            InstallerVersion='450'
+            InstallerVersion='500'
             Languages='1033'
             Compressed='yes'
             InstallScope='perMachine'
-            SummaryCodepage='1252'
-            Platform='$(var.Platform)'/>
```
Comment (chriskrycho) [inline wix/main.wxs:52]: Why are you dropping the `Platform` value here?

File: wix/main.wxs [hunk 6]
```diff
@@ -62,16 +59,18 @@
                 <Directory Id='INSTALLDIR' Name='Volta'>
                 </Directory>
             </Directory>
+            <?if $(sys.BUILDARCH) = x64 ?>
             <Merge Id='VCRedist' SourceFile='wix\Microsoft_VC140_CRT_x64.msm' DiskId='1' Language='0'/>
+            <?endif ?>
```
Comment (chriskrycho) [inline wix/main.wxs:64]: Why are you dropping this check?

File: wix/main.wxs [hunk 7]
```diff
@@ -62,16 +59,18 @@
...
         </Directory>
 
         <ComponentGroup Id='Binaries' Directory='INSTALLDIR'>
-            <Component Id='voltaBinary' Guid='*' Win64='$(var.Win64)'>
+            <Component Id='voltaBinary' Guid='*'>
```
Comment (chriskrycho) [inline wix/main.wxs:68]: Here and throughout, why are you removing `Win64=`?

**Review by chriskrycho — COMMENTED** [review]
This looks reasonable to me overall! I had a bunch of questions, but they are questions where I literally just do not understand the reasoning, so do not take them as implying you’re *wrong* in making those changes—merely that I would like to understand them and make sure we have the reason for the change captured for posterity!

Also, @charlespierce can you take a gander here?

File: wix/main.wxs (hunk 4 above)
Comment (shibayan) [reply wix/main.wxs:198]: Added a condition that the VC++ redistribution runtime should not be included on arm64 since it was included for x64.

However, I do not know why the redistribution runtime is included. In my opinion, it seems unnecessary.

File: wix/main.wxs (hunk 3 above)
Comment (shibayan) [reply wix/main.wxs:28]: The `Win64` and `Platform` settings have been removed as they are now deprecated in wix. The details are described in the following pull request from cargo-wix, which is shared here.

https://github.com/volks73/cargo-wix/pull/113#issue-723263542

>I took the opportunity to remove use of Win64 and Platform in the wxs: those attributes are deprecated according to the wix documentation, passing -arch on the command line should be prefered (which this PR now does). var.Platform is replaced by sys.BUILDARCH.

File: wix/main.wxs (hunk 5 above)
Comment (shibayan) [reply wix/main.wxs:52]: I am deleting it for the same reason as here: https://github.com/volta-cli/volta/pull/1801#discussion_r1687443667

File: wix/main.wxs (hunk 7 above)
Comment (shibayan) [reply wix/main.wxs:68]: I am deleting it for the same reason as here: https://github.com/volta-cli/volta/pull/1801#discussion_r1687443667

File: wix/main.wxs (hunk 6 above)
Comment (shibayan) [reply wix/main.wxs:64]: Since the VC++ redistribution runtime was only available for x64, I have added a condition so that it is only included for x64 as before.

File: wix/main.wxs (hunk 4 above)
Comment (charlespierce) [reply wix/main.wxs:198]: It looks like that was added in #592, as it's needed to actually run Volta and not always available on the system.

**Review by charlespierce — APPROVED** [review]
LGTM! We've never actually supported / done builds for 32-bit Windows anyway, so the removal of the 64-bit checks is even more welcome!

File: wix/main.wxs (hunk 4 above)
Comment (shibayan) [reply wix/main.wxs:198]: I looked into it because I couldn't find a merge module for Arm64, and it seems that it is now deprecated.

>Merge modules (.msm files) for Visual C++ Redistributable files are deprecated. We don't recommend you use them for application deployment. Instead, we recommend central deployment of the Visual C++ Redistributable package.

https://learn.microsoft.com/en-us/cpp/windows/redistributing-visual-cpp-files?view=msvc-170#install-the-redistributable-merge-modules

**Comment by chriskrycho** [comment]
Excellent – and @shibayan answered all my questions (thank you!).
